deal/introspection/_extractor.py

```diff
@@ -52,6 +62,6 @@ def get_contracts(func: Callable) -> Iterator[Contract]:
             if contracts.patcher:
                 yield _wrappers.Has(contracts.patcher)
 
-        if not hasattr(func, '__wrapped__'):
+        if not has_wrapped(func):
```

Comment:
Not much value in this, tbh: the `__wrapped__` is used the very next line. While I like mypy with all my heart, type safety makes more sense when it brings checks not only to the very next line.

Reply:
Okay, got it! Maybe then add `# type: ignore` on this line?